### PR TITLE
ensure profitability && pure arbitrage in jupiter pingpong

### PR DESCRIPTION
**Pull request by staccDOTsol**
image

### CODE REVIEW

**Comment by staccDOTsol** [comment]
Jarett Dunn, [1/11/23 4:02 AM]
1. ensure profitability

Jarett Dunn, [1/11/23 4:02 AM]
it sends a tx to itself after the arb

Jarett Dunn, [1/11/23 4:02 AM]
for the og amount of whatever tokens it's trading

Jarett Dunn, [1/11/23 4:02 AM]
if it would have lost $

Jarett Dunn, [1/11/23 4:02 AM]
it fails

Jarett Dunn, [1/11/23 4:02 AM]
atomicaly

Jarett Dunn, [1/11/23 4:02 AM]
2. it does pure arbitrage in pingpong

Jarett Dunn, [1/11/23 4:02 AM]
so it's tokena -> tokenb -> tokena

Jarett Dunn, [1/11/23 4:02 AM]
without the risk and exposure of holding a token and trying to swap back

Jarett Dunn, [1/11/23 4:02 AM]
either of these features is significantly better than the og code

**Comment by alexonab** [comment]
Testing so far shows:

1. the GUI does not load past setup
2. the arbitrage strat fails to execute a forced trade on both Jup and Prism

```
[E] PRESSED - EXECUTION FORCED BY USER!
Swap error:  TypeError: Cannot read properties of undefined (reading 'marketInfos')
    at _i.exchange (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/node_modules/@jup-ag/core/dist/core.cjs.production.min.js:1:176039)
    at swap (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/swap.js:85:22)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async arbitrageStrategy (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:395:30)
    at async watcher (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:460:4)
TypeError: Cannot read properties of undefined (reading 'outputAmount')
    at arbitrageStrategy (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:400:61)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async watcher (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:460:4)

[E] PRESSED - EXECUTION FORCED BY USER!
TypeError: user.signAllTransactions is not a function
    at /home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/node_modules/@prism-hq/prism-ag/dist/swap/swap.js:236:38
    at Generator.next (<anonymous>)
    at fulfilled (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/node_modules/@prism-hq/prism-ag/dist/swap/swap.js:5:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
undefined
Swap error:  TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
    at Object.writeFileSync (node:fs:2211:5)
    at storeItInTempAsJSON (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/utils/index.js:9:5)
    at swap (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/swap.js:158:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async arbitrageStrategy (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:395:30)
    at async watcher (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:460:4) {
  code: 'ERR_INVALID_ARG_TYPE'
}
TypeError: Cannot read properties of undefined (reading 'outputAmount')
    at arbitrageStrategy (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:400:61)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async watcher (/home/sysadmin/solana-jupiter-bot/dev/p/solana-jupiter-bot-features-prism/src/bot/index.js:460:4)`

**Comment by staccDOTsol** [comment]
```user.signAllTransactions is not a function``` 

this is interesting, there appears to be a dependency issue

**Comment by staccDOTsol** [comment]
you were right about prism, long video uploads

**Comment by staccDOTsol** [comment]
tl;dr works fine, :) sorry I didn't elaborate but the off-git conversations as well as the commit 'ui sux' meant that I gutted the ui. that includes hotkeys. for the time being :)
image

**Comment by staccDOTsol** [comment]
https://drive.google.com/file/d/1kwzGcVJYBMc79YS97X1xVqGqgs-i-5Ho/view?usp=sharing
